portal-web/docroot/html/taglib/ui/map/page.jsp

```diff
@@ -19,61 +19,68 @@
 <%
 String protocol = HttpUtil.getProtocol(request);
 
-String apiKey = GetterUtil.getString(request.getAttribute("liferay-ui:map:apiKey"));
 boolean geolocation = GetterUtil.getBoolean(request.getAttribute("liferay-ui:map:geolocation"));
 double latitude = (Double)request.getAttribute("liferay-ui:map:latitude");
 double longitude = (Double)request.getAttribute("liferay-ui:map:longitude");
 String name = GetterUtil.getString((String)request.getAttribute("liferay-ui:map:name"));
 String points = GetterUtil.getString(request.getAttribute("liferay-ui:map:points"));
-String provider = GetterUtil.getString((String)request.getAttribute("liferay-ui:map:provider"));
 
-if (Validator.isNull(provider)) {
-	Group group = layout.getGroup();
+PortletPreferences companyPortletPreferences = PrefsPropsUtil.getPreferences(company.getCompanyId());
 
-	provider = group.getLiveParentTypeSettingsProperty("provider");
+String googleMapsAPIKey = PrefsParamUtil.getString(companyPortletPreferences, request, "googleMapsAPIKey", "");
```

Comment:
We shouldn't be using PrefsParamUtil here because we don't want to read the value from the request. We should either read it from the taglib attribute or from the preferences, but not from the request.